### PR TITLE
[JavaToolInstallerV0, NodeToolV0] Added restrictions

### PR DESCRIPTION
**Pull request by sdobrodeev**
**Task name**: JavaToolInstallerV0, NodeToolV0

**Description**: Added restrictions

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/JavaToolInstallerV0/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -21,7 +21,7 @@
   "loc.input.label.azureCommonVirtualFile": "Common virtual path",
   "loc.input.help.azureCommonVirtualFile": "Path to the JDK inside the Azure storage container.",
   "loc.input.label.jdkDestinationDirectory": "Destination directory",
-  "loc.input.help.jdkDestinationDirectory": "On Linux and Windows, this is used as the destination directory for JDK installation. On macOS, this directory is used as a temporary folder for extracting .dmg's since macOS doesn't support installing of JDK to specific directory.",
+  "loc.input.help.jdkDestinationDirectory": "On Linux and Windows, this is used as the destination directory for JDK installation. On macOS, this directory is used as a temporary folder for extracting of .dmg's since macOS doesn't support installing of JDK to specific directory.",
```
Comment (DaniilShmelev) [inline Tasks/JavaToolInstallerV0/Strings/resources.resjson/en-US/resources.resjson:24]: Could you please clarify why this change was made?

**Review by DaniilShmelev — APPROVED** [review]

File: Tasks/JavaToolInstallerV0/task.json [hunk 2]
```diff
@@ -148,6 +149,14 @@
             "argumentFormat": ""
         }
     },
+    "restrictions": {
+        "commands": {
+            "mode": "restricted"
+        },
+        "settableVariables": {
+            "allowed": ["JAVA_HOME_*"]
```
Comment (anatolybolshakov) [inline Tasks/JavaToolInstallerV0/task.json:157]: This task could update JAVA_HOME variable also - could you please add it?

Comment (anatolybolshakov) [inline Tasks/JavaToolInstallerV0/task.json:157]: JavaToolInstaller also calls toolLib.prependPath, which updates PATH variables - so I guess we also need to put it here.

File: Tasks/NodeToolV0/task.loc.json [hunk 3]
```diff
@@ -55,7 +56,15 @@
       "argumentFormat": ""
     }
   },
+  "restrictions": {
+    "commands": {
+      "mode": "restricted"
+    },
+    "settableVariables": {
+      "allowed": []
```
Comment (anatolybolshakov) [inline Tasks/NodeToolV0/task.loc.json:64]: This task also updates PATH via toolLib.prependPath

**Review by anatolybolshakov — APPROVED** [review]
